specification/consumption/resource-manager/Microsoft.Consumption/stable/2022-06-01/consumption.operations.pricesheet.json

```diff
@@ -0,0 +1,383 @@
+    "description": "Consumption management client provides access to consumption resources for Azure Enterprise Subscriptions."
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "securityDefinitions": {
+    "azure_auth": {
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow.",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/providers/Microsoft.Consumption/pricesheets/default": {
+      "get": {
+        "tags": [
+          "PriceSheet"
+        ],
+        "operationId": "PriceSheet_List",
+        "description": "List the price sheet for a subscription. Price sheet is available via this API only for May 1, 2014 or later.",
+        "externalDocs": {
+          "url": "https://docs.microsoft.com/en-us/rest/api/consumption/"
+        },
+        "x-ms-examples": {
+          "PriceSheet": {
+            "$ref": "./examples/PriceSheet.json"
+          }
+        },
+        "parameters": [
+          {
+            "name": "$expand",
+            "description": "May be used to expand the properties/meterDetails within a price sheet. By default, these fields are not included when returning price sheet.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$skiptoken",
+            "description": "Skiptoken is only used if a previous operation returned a partial result. If a previous response contains a nextLink element, the value of the nextLink element will include a skiptoken parameter that specifies a starting point to use for subsequent calls.",
+            "in": "query",
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "$top",
+            "description": "May be used to limit the number of results to the top N results.",
+            "in": "query",
+            "required": false,
+            "type": "integer",
+            "format": "int32",
+            "minimum": 1,
+            "maximum": 1000
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v3/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/PriceSheetResult"
```

Comment:
>PriceSheetResult [](http://example.com/codeflow?start=37&length=16)

This is listing API but the response is not in array form? Looks like in your example, the response is in array form

Reply:
![image](https://user-images.githubusercontent.com/72947097/208208037-b86e6a25-4f5d-4a79-80a8-90e27b1f8e39.png)

The response property pricesheets items are array.

Reply:
Good point. When you implement a LIST api, ARM expects you to return an object with a 'value' property that holds the actual array of resources/objects.

Reply:
But... this isn't actually a LIST api, in the resource sense - in terms of the request path, its just a singleton get, i.e. fetching ONE pricesheet named default.

Reply:
And that resource just happens to contain a list....

Reply:
I am unclear why its this way and its not a normal ARM LIST api, actually.

Reply:
Agreed. I can take it up with my team to understand more on this, but we cannot change a response model without letting clients know right? They will have dependency on existing model. What do you suggest? 
This response model is same for all other versions of this API

Reply:
@TimLovellSmith Can you please help on how should we proceed forward with this?

Reply:
@jayati-tehri The answers to this depends on how you want to model this API overall : 

For example the API 
GET /subscriptions/{subscriptionId}/providers/Microsoft.Consumption/pricesheets/default

Currently this is modelled as a single point GET of a subscription level pricesheet resource called default. Which means it should return only one resource and should not have nextlink. If you want to model this as a list instead to list all the individual pricesheets in the subscription then the call needs to be like this : 

GET /subscriptions/{subscriptionId}/providers/Microsoft.Consumption/pricesheets/

and this returns the list of various pricesheets in the subscription in the array format with next link.


Also there is an issue with how you have created the response for a list operation , the nextlink should not be inside properties , check the output of the operationsList.json where you have modelled this correctly. This is explained in the microsoft REST guidelines here : https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md#collections

You should also go over that document in entirety to model these apis correctly. 

Please fix these APIs and let us know , if you need any further clarifications , please feel free to send me and the apireviews an email